tests/phpunit/unit/api/actions/beansGetCurrentAction.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Tests for _beans_get_current_action()
+ *
+ * @package Beans\Framework\Tests\Unit\API\Actions
+ *
+ * @since   1.5.0
+ */
+
+namespace Beans\Framework\Tests\Unit\API\Actions;
+
+use Beans\Framework\Tests\Unit\Test_Case;
+
+/**
+ * Class Tests_BeansGetCurrentAction
+ *
+ * @package Beans\Framework\Tests\Unit\API\Actions
+ * @group   unit-tests
+ * @group   api
+ */
+class Tests_BeansGetCurrentAction extends Test_Case {
+
+	/**
+	 * Setup test fixture.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once BEANS_TESTS_LIB_DIR . 'api/actions/functions.php';
+		require_once BEANS_TESTS_LIB_DIR . 'api/utilities/functions.php';
+	}
+
+	/**
+	 * Reset the test fixture.
+	 */
+	protected function tearDown() {
+		parent::tearDown();
+
+		global $_beans_registered_actions;
+		$_beans_registered_actions = array(
+			'added'    => array(),
+			'modified' => array(),
+			'removed'  => array(),
+			'replaced' => array(),
+		);
+	}
+
+	/**
+	 * Test _beans_get_current_action() should return false when the ID is registered with the "removed" status.
+	 */
+	public function test_should_return_false_when_removed_status() {
+		global $_beans_registered_actions;
+		$_beans_registered_actions['removed']['foo'] = wp_json_encode( array(
+			'hook'     => 'foo',
+			'callback' => 'callback',
+			'priority' => 10,
+			'args'     => 1,
+		) );
+
+		$this->assertFalse( _beans_get_current_action( 'foo' ) );
+	}
+
+	/**
+	 * Test _beans_get_current_action() should return false when the ID is registered with the "removed" status.
```

Comment:
Identical comment with previous test. 
Should it read: `return false when the action is invalid` ?

Reply:
@christophherr No, it's testing the "removed" status.  Look on line 53, where we first store it in the "removed" status.  That registers it with Beans.  

Then if the `$id` exists in the `'removed'` status, the function bails out, returning `false`.